monitoring/cgroup.go

```diff
@@ -46,35 +49,38 @@ type cgroupChecker struct {
 
 // Name returns name of the checker
 // Implements health.Checker
-func (r cgroupChecker) Name() string {
+func (c *cgroupChecker) Name() string {
 	return cgroupCheckerID
 }
 
-// Check verifies existence of cgroup mounts given in r.cgroups.
+// Check verifies existence of cgroup mounts given in c.cgroups.
 // Implements health.Checker
-func (r cgroupChecker) Check(ctx context.Context, reporter health.Reporter) {
-	var probes health.Probes
-	err := r.check(ctx, &probes)
-	if err != nil && !trace.IsNotFound(err) {
-		reporter.Add(NewProbeFromErr(r.Name(), "failed to validate cgroup mounts", err))
-		return
+func (c *cgroupChecker) Check(ctx context.Context, reporter health.Reporter) {
+	probes, err := c.check(ctx)
+	if err != nil {
+		probes.Add(NewProbeFromErr(c.Name(), "failed to validate cgroup mounts", err))
 	}
+	health.AddFrom(reporter, probes)
+}
 
-	health.AddFrom(reporter, &probes)
-	if probes.NumProbes() != 0 {
-		return
-	}
+// check verifies that all expected cgroups have been mounted. Skips check if
+// mounts file is not available.
+func (c *cgroupChecker) check(ctx context.Context) (probes health.Reporter, err error) {
+	probes = &health.Probes{}
 
-	reporter.Add(NewSuccessProbe(r.Name()))
-}
+	mounts, err := c.getMounts(ctx)
+
+	// Skip check if mounts file is not available
+	if trace.IsNotFound(err) {
+		probes.Add(NewSuccessProbe(c.Name()))
```

Comment:
Isn't that a bit misleading that we're adding "success" probe if the check hasn't, in fact, run? As opposed to just not adding any kind of probe. Was it the same way before?

Reply:
https://github.com/gravitational/satellite/blob/de06a5d14d20373e8296f84897f624c2fdaaacb8/monitoring/cgroup.go#L53-L94
It looks like that is what we had before.
`check(...)` only adds failed probes to the reporter.
If no probes were reporter we add a success probe.